src/XamlX.IL.Cecil/CecilEmitter.cs

```diff
@@ -155,7 +155,18 @@ public IXamlILEmitter Emit(SreOpCode code, float arg)
             public IXamlILEmitter Emit(SreOpCode code, double arg)
                 => Emit(Instruction.Create(Dic[code], arg));
 
+            private MethodReference ImportReference(MethodReference r)
+            {
+                var rv = M.ImportReference(r);
+
+                if (r.DeclaringType is GenericInstanceType gi)
```

Comment:
1) Probably need to do that recursively
2) I'd prefer to not mutate the passed DeclaringType, it might break something elsewhere.